bins/reth-crawler/src/api_server/peerdb/app_state.rs

```diff
@@ -0,0 +1,23 @@
+use std::sync::Arc;
+
+use axum::extract::FromRef;
+use reth_crawler_db::{db::SqlPeerDB, AwsPeerDB, PeerDB};
+
+#[derive(Clone, FromRef)]
+pub struct AppState {
+    store: Arc<dyn PeerDB>,
```

Comment:
should be possible to make `AppState` here (and the rest of the references) generic over the PeerDB trait